Tasks/Gradle/CodeAnalysis/Common/CodeAnalysisResultPublisher.ts

```diff
@@ -24,8 +24,20 @@ export class CodeAnalysisResultPublisher {
      *
      * @param {string} prefix - used to discriminate between artifacts comming from different builds of the same projects (e.g. the build number)
      */
-    public uploadArtifacts(prefix: string):void {
+    public uploadArtifacts(prefix: string):number {
+        // If there are no results to upload, return
         if (this.analysisResults.length === 0) {
+            tl.debug("Skipping artifact upload: No analysis results");
+            return;
+        }
+
+        // If there are no files to upload, return
+        var analysisResultsWithFiles:AnalysisResult[] = this.analysisResults.filter(
```

Comment:
Using Javascript feature here - _filter()_ creates a new array with only the elements for which the callback returns true. The callback passed below does a null/undefined check for safety, then returns true if there were files to upload.